specification/cdn/resource-manager/Microsoft.Cdn/stable/2020-09-01/afdx.json

```diff
@@ -0,0 +1,5573 @@
+            "in": "path",
+            "description": "Name of the domain under the profile which is unique globally.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/AFDDomain"
+            }
+          },
+          "default": {
+            "description": "CDN error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/AfdErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "AFDCustomDomains"
+        ],
+        "description": "Creates a new domain within the specified profile.",
+        "operationId": "AFDCustomDomains_Create",
+        "x-ms-examples": {
+          "AFDCustomDomains_Create": {
+            "$ref": "./examples/AFDCustomDomains_Create.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "name": "profileName",
+            "in": "path",
+            "description": "Name of the CDN profile which is unique within the resource group.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "customDomainName",
+            "in": "path",
+            "description": "Name of the domain under the profile which is unique globally",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "customDomain",
+            "in": "body",
+            "description": "Domain properties",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AFDDomain"
+            }
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/AFDDomain"
+            }
+          },
+          "201": {
+            "description": "Created. The request has been fulfilled and a new domain has been created.",
+            "schema": {
+              "$ref": "#/definitions/AFDDomain"
+            }
+          },
+          "default": {
+            "description": "CDN error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/AfdErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
```

Comment:
For async PUT, `final-state-via` should be either `original-uri` (when `ProvisioningState` can be tracked by the original PUT URI) or `azure-async-operation` (when the response provides `Azure-AsyncOperation `header). Can you please update accordingly for all async PUTs?

https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creatingupdating-using-put

https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation-options

Reply:
All changed to azure-async-operation